src/ds5/ds5-factory.cpp

```diff
@@ -956,6 +958,43 @@ namespace librealsense
 
             return tags;
         };
+
+        std::shared_ptr<safety_preset> get_safety_preset_at_index(int index) override
+        {
+            std::shared_ptr<safety_preset> result;
+            
+            // prepare command
+            command cmd(ds::SAFETY_PRESET_READ);
+            cmd.require_response = false;
```

Comment:
No response on `get`?
I am not very familiar with this mechanism but we do need response, right?